core/types/qi_tx.go

```diff
@@ -35,6 +36,7 @@ func (tx *QiTx) copy() TxData {
 		ParentHash: tx.ParentHash,
 		MixHash:    tx.MixHash,
 		WorkNonce:  tx.WorkNonce,
+		Data:       common.CopyBytes(tx.Data),
```

Comment:
arent there Qi test cases you need to change to test that this is working fine?

Reply:
Fixed